libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -307,10 +307,11 @@ bool ReferencesResolver::visit(Return const& _return)
 
 void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 {
+	bool isStateVariable = dynamic_cast<VariableDeclaration const*>(&_variable);
```

Comment:
```suggestion
	bool isStateVariable = dynamic_cast<StateVariableDeclaration const*>(&_variable);
```
?

Reply:
That's a defect ;-) Thanks!

Reply:
It's interesting compiler errors in CI: ``the compiler can assume that the address of ‘_variable’ will never be NULL`` - never seen those, but it's legit :-)!